### PR TITLE
agent: auto-online hotplug memory

### PR DESCRIPTION
**Pull request by bergwolf**
Kata agent does not wait for memory hotplug, and kata-runtime
does not wait on cpu/memory online. As a temporary fix, let's
make sure memory are onlined by the uevent listener at least.

Fixes: #363

### CODE REVIEW

**Comment by gnawux** [comment]
nice
LGTM

**Comment by devimc** [comment]
cc @clarecch  @linzichang

File: agent.go [hunk 1]
```diff
@@ -486,6 +486,11 @@ func (s *sandbox) listenToUdevEvents() {
 			continue
 		}
 
+		// We only care about add event
+		if uEv.Action != "add" {
+			continue
+		}
```
Comment (sboeuf) [inline agent.go]: Please put this in a separate commit, as this is not related to the uevent for memory.

File: agent.go [hunk 2]
```diff
@@ -495,7 +500,7 @@ func (s *sandbox) listenToUdevEvents() {
 		})
 
 		// Check if device hotplug event results in a device node being created.
-		if uEv.DevName != "" && uEv.Action == "add" && strings.HasPrefix(uEv.DevPath, rootBusPath) {
+		if uEv.DevName != "" && strings.HasPrefix(uEv.DevPath, rootBusPath) {
```
Comment (sboeuf) [inline agent.go]: Same thing here, this should go to a separate commit.

**Comment by linzichang** [comment]
I have a question. Will this face the same question I found at https://github.com/kata-containers/runtime/pull/624#issuecomment-419351740 ?

File: agent.go (hunk 1 above)
Comment (bergwolf) [reply agent.go]: Why ask for splitting a commit into two? The checks **MUST** be separated otherwise I have to check it again when testing for memory online path.

File: agent.go (hunk 2 above)
Comment (bergwolf) [reply agent.go]: Same here. It **IS** a single commit.

**Comment by bergwolf** [comment]
@linzichang Are you referring to the systemd udev issue? For one thing, the commit does not make things worse and it actually fixes the `-m` memory hotplug issue I'm seeing with vm factory. For another thing, I think systemd udev should be disabled, otherwise the uevent listener with device map approach is broken, ditto the cpu online waiting logic.

**Comment by linzichang** [comment]
@bergwolf The main issue is `onlineCPUResources` conflicting with other online method. See https://github.com/kata-containers/agent/blob/master/grpc.go#L145 . Your changes just care about memory, so it wouldn't be a problem :) But the `udev` should be disabled.

File: agent.go (hunk 1 above)
Comment (sboeuf) [reply agent.go]: Ah ok I understand.

**Comment by sboeuf** [comment]
Travis is stuck as for a bunch of other PRs. Ignoring for now and merging this
